clients/src/main/java/org/apache/kafka/common/utils/Timer.java

```diff
@@ -172,6 +172,17 @@ public long currentTimeMs() {
         return currentTimeMs;
     }
 
+    /**
+     * Get the deadline time in milliseconds when this timer would expire. The deadlineMs time
+     * is generally updated through {@link #reset(long)} method or {@link #updateAndReset(long)}
+     * method call.
+     *
+     * @return The deadline time in milliseconds at which the timer would expire.
+     */
+    public long deadlineMs() {
+        return deadlineMs;
+    }
+
```

Comment:
This method seems to only be useful in the test assertions, and that doesn't seem like a good enough reason to expose this functionality in the common utils.

Reply:
hmm maybe it wasn't needed before? I see no difference in exposing this v/s exposing `currentTimeMs` for example. Let me know what you think about it?